AppController/djinn.rb

```diff
@@ -3823,10 +3828,18 @@ def build_search_service2
 
   # Run a build on modified directories so that changes will take effect.
   def build_uncommitted_changes
-    status = `git -C #{APPSCALE_HOME} status`
+    if @options['update'].nil?
+      return
+    end
+
+    update_dir = @options['update']
+
+    if update_dir == "all"
+      update_dir = ALLOWED_DIR_UPDATES.join(',')
```

Comment:
This should probably be plural (`update_dirs`), and you could assign `ALLOWED_DIR_UPDATES` as is without joining the elements.